src/main/java/com/exadel/fedorov/orders/service/OrderService.java

```diff
@@ -1,42 +1,104 @@
+        Order order = orderDAO.findById(id);
+        List<OrderDetail> items = orderDAO.findItemsByOrderId(id);
 
-    public Optional<Order> findById(Long id) {
-        return orderDAO.findById(id);
+        return Optional.of(fillRespOrderDto(order, items));
     }
 
-    public void update(Order order) {
-        orderDAO.update(order);
+    public void update(Long id, String status, String statusDescription) {
+        orderDAO.update(id, OrderStatus.valueOf(status), statusDescription);
     }
 
     public void deleteById(Long id) {
         orderDAO.deleteById(id);
     }
 
-    public void createOrder(ReqOrderDTO reqOrderDTO) {
-        orderDAO.createOrderWithProcedure(reqOrderDTO);
+    public void createOrder(List<ReqOrderItemDTO> orderItems, String clientName, String statusDescription) {
+        Order order = new Order(clientName, getTotalPrice(orderItems), statusDescription);
+
+        List<OrderDetail> details = convertReqOrderItemDTOsToOrderDetails(orderItems);
+        orderDAO.createOrderRecordWithProcedure(order, details);
+    }
+
+
+    private BigDecimal getTotalPrice(List<ReqOrderItemDTO> orderItems) {
+        BigDecimal total = new BigDecimal(ZERO);
+        for (ReqOrderItemDTO item : orderItems) {
+            total = total.add(item.getPrice());
+        }
+        return total;
+    }
+
+    private List<RespOrderDTO> convertOrdersToRespOrderDTOs(List<Order> orders) {
+        List<RespOrderDTO> orderDTOS = new ArrayList<>();
+        for (Order order : orders) {
+            List<OrderDetail> details = orderDAO.findItemsByOrderId(order.getId());
+            orderDTOS.add(fillRespOrderDto(order, details));
+        }
+        return orderDTOS;
+    }
+
+    private RespOrderDTO fillRespOrderDto(Order order, List<OrderDetail> details) {
+        return new RespOrderDTO(
+                convertOrderDetailsToRespOrderItemDTOs(details),
+                order.getTotalPrice(),
+                order.getClientName(),
+                order.getStatus().toString(),
+                order.getStatusDescription(),
+                order.getTime().toLocalDateTime());
+    }
+
+    private List<RespOrderItemDTO> convertOrderDetailsToRespOrderItemDTOs(List<OrderDetail> details) {
+        return details.stream()
+//                .map(OrderItemMapper.INSTANCE::itemToRespItemDto)
+                .map(this::convertOrderDetailsToRespItemDto)
+                .collect(Collectors.toList());
+    }
+
+    private RespOrderItemDTO convertOrderDetailsToRespItemDto(OrderDetail orderDetail) {
+        return new RespOrderItemDTO(orderDetail.getProductId(), orderDetail.getPositionPrice(), orderDetail.getProductCount());
+    }
+
+    private List<OrderDetail> convertReqOrderItemDTOsToOrderDetails(List<ReqOrderItemDTO> dtoList) {
+        return dtoList.stream()
+                .map(this::convertReqItemDtoToOrderDetail)
+//                .map(OrderItemMapper.INSTANCE::reqItemDtoToItem)
```

Comment:
Why don't you use map struct here?

Reply:
Map struct does not work (spring can not find implementation of bean)
Did it with  model mapper